crates/nu-command/src/commands/filters/select.rs

```diff
@@ -119,7 +123,11 @@ fn select(args: CommandArgs) -> Result<OutputStream, ShellError> {
                         return Err(reason);
                     }
 
-                    bring_back.entry(key.clone()).or_insert(vec![]);
+                    // No value for column 'key' found, insert nothing to make sure all rows contain all keys.
+                    bring_back
+                        .entry(key.clone())
+                        .or_insert(vec![])
+                        .push(UntaggedValue::nothing().into());
```

Comment:
This is the actual fix. Not only create the empty vector for given column but also push `nothing`.